### ISSUE TITLE
Panache IN-query errors for BigDecimals: `Binding is multi-valued; illegal call to #getBindValue`

### ISSUE THREAD

**Issue by Felk** [quarkusio/quarkus#38276]
### Describe the bug

Starting with Quarkus 3.7.0.CR1, Panache throws an exception trying to parse an IN-query that has a multi-valued set of BigDecimals. The error message is `Binding is multi-valued; illegal call to #getBindValue`. The error does not occurr for e.g. strings, and it does not occurr in version 3.6.6.

### Expected behavior

No error, successful IN-query execution.

### Actual behavior

```
Caused by: java.lang.IllegalStateException: Binding is multi-valued; illegal call to #getBindValue
	at org.hibernate.query.internal.QueryParameterBindingImpl.getBindValue(QueryParameterBindingImpl.java:96)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.resolveSqmParameter(BaseSqmToSqlAstConverter.java:6053)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.consumeSqmParameter(BaseSqmToSqlAstConverter.java:5654)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.consumeSingleSqmParameter(BaseSqmToSqlAstConverter.java:5743)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.processInSingleParameter(BaseSqmToSqlAstConverter.java:7919)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.processInSingleHqlParameter(BaseSqmToSqlAstConverter.java:7869)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.processInListWithSingleParameter(BaseSqmToSqlAstConverter.java:7858)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.visitInListPredicate(BaseSqmToSqlAstConverter.java:7765)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.visitInListPredicate(BaseSqmToSqlAstConverter.java:440)
	at org.hibernate.query.sqm.tree.predicate.SqmInListPredicate.accept(SqmInListPredicate.java:147)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.visitWhereClause(BaseSqmToSqlAstConverter.java:2484)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.visitQuerySpec(BaseSqmToSqlAstConverter.java:2061)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.visitQuerySpec(BaseSqmToSqlAstConverter.java:440)
	at org.hibernate.query.sqm.tree.select.SqmQuerySpec.accept(SqmQuerySpec.java:125)
	at org.hibernate.query.sqm.spi.BaseSemanticQueryWalker.visitQueryPart(BaseSemanticQueryWalker.java:218)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.visitQueryPart(BaseSqmToSqlAstConverter.java:1915)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.visitSelectStatement(BaseSqmToSqlAstConverter.java:1600)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.visitSelectStatement(BaseSqmToSqlAstConverter.java:440)
	at org.hibernate.query.sqm.tree.select.SqmSelectStatement.accept(SqmSelectStatement.java:228)
	at org.hibernate.query.sqm.sql.BaseSqmToSqlAstConverter.translate(BaseSqmToSqlAstConverter.java:776)
	at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.buildCacheableSqmInterpretation(ConcreteSqmSelectQueryPlan.java:399)
	at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.withCacheableSqmInterpretation(ConcreteSqmSelectQueryPlan.java:324)
	at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.performList(ConcreteSqmSelectQueryPlan.java:300)
	at org.hibernate.query.sqm.internal.QuerySqmImpl.doList(QuerySqmImpl.java:509)
	at org.hibernate.query.spi.AbstractSelectionQuery.list(AbstractSelectionQuery.java:427)
	at org.hibernate.query.Query.getResultList(Query.java:120)
	at io.quarkus.hibernate.orm.panache.common.runtime.CommonPanacheQueryImpl.list(CommonPanacheQueryImpl.java:280)
	at io.quarkus.hibernate.orm.panache.runtime.PanacheQueryImpl.list(PanacheQueryImpl.java:149)
	at org.acme.ReproducerTest.lambda$illegalBindingMultiValuedReproducer$0(ReproducerTest.java:20)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:71)
	... 18 more
```

### How to Reproduce?

TestEntity.java
```java
@Entity
public class TestEntity extends PanacheEntity {
    public BigDecimal bigDecimalField;
}
```

ReproducerTest.java
```java
@QuarkusTest
@TestTransaction
class ReproducerTest {
    @Test
    void illegalBindingMultiValuedReproducer() {
        Set<BigDecimal> multiValuedParameter = Set.of(BigDecimal.ONE, BigDecimal.TEN);
        assertDoesNotThrow(() -> TestEntity
                .find("bigDecimalField IN (?1)", multiValuedParameter)
                .list());
    }
}
```

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19045.3803]

### Output of `java -version`

openjdk version "21" 2023-09-19
OpenJDK Runtime Environment (build 21+35-2513)
OpenJDK 64-Bit Server VM (build 21+35-2513, mixed mode, sharing)

### Quarkus version or git rev

3.7.0.CR1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

PS C:\rd\spot> mvn --version
Apache Maven 3.9.3 (21122926829f1ead511c958d89bd2f672198ae9f)
Maven home: C:\Program Files\maven\apache-maven-3.9.3
Java version: 21, vendor: Oracle Corporation, runtime: C:\Program Files\OpenJDK\jdk-21
Default locale: de_DE, platform encoding: UTF-8
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### Additional information

Reproducer project: 
[quarkus-illegal-multi-valued-reproducer.zip](https://github.com/quarkusio/quarkus/files/13974871/quarkus-illegal-multi-valued-reproducer.zip)

**Comment by sahlex** [comment]
Hi @Felk. I had the same issue when using spring 3.2 with Hibernate 6.4.1.Final.
The bug does seem to have been fixed in 6.4.2.Final as it's working fine for me.

**Comment by Felk** [comment]
Thanks for the hint. I can confirm it works with the nightly build, so this is fixed by https://github.com/quarkusio/quarkus/pull/38283
